cmd/containerd-shim-runc-v2/task/service.go

```diff
@@ -651,9 +667,33 @@ func (s *service) send(evt interface{}) {
 	s.events <- evt
 }
 
+func (s *service) handleProcessExitL(e runcC.Exit, c *runc.Container, p process.Process) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	s.handleProcessExit(e, c, p)
+}
+
 // s.mu must be locked when calling handleProcessExit
```

Comment:
[No, it does not.](https://github.com/containerd/containerd/pull/9104#issuecomment-1737902483) I'm probably not going to win the argument in this PR, but that won't stop me from reminding everyone in every PR which touches this file.

Reply:
Oh! I've missed and/or forgot that discussion. Yes, this was something that looked off to me again while working on this but I got distracted with other things and ended up forgetting about it. I'd rather not block this change on that (apparently contentious) topic, but I'd be happy to open another PR so we can revisit this lock.